crypto/secp256k1/secpk256k1_test.go

```diff
@@ -63,3 +65,15 @@ func TestSignAndValidateSecp256k1(t *testing.T) {
 
 	assert.False(t, pubKey.VerifyBytes(msg, sig))
 }
+
+func TestSecp256k1LoadPrivkeyAndSerializeIsIdentity(t *testing.T) {
```

Comment:
This test is cool to show the reviewer that what you are doing is OK. Not sure, if in two years from now people will wonder why it is included. Maybe add a comment why this is useful.

Also, did you experiment with an edge case: What if the random bytes already contain zeroes? I think the padding was partly to make sure that leading zeroes in `pk` be "ignored" by `new(big.Int).SetBytes(pk)` in https://github.com/btcsuite/btcd/blob/fb90c334dffc6c91843d8d36ac31e7f7eb6c7594/btcec/privkey.go#L21-L31
This might be even less likely when using the output of sha256. But it would be great that this didn't change with this PR.

Reply:
Clarified this with @ValarDragon on slack. This isn't an issue here, as `Serialize` is still called.